File: 2020/day22/1.py
from collections import deque

file = open("input.txt", "r")
lines = file.read()
file.close()

def main(rawInput):
	player1, player2 = rawInput.split('\n\n')
	player1 = deque(map(int, player1.split('\n')[1:]))
	player2 = deque(map(int, player2.split('\n')[1:]))
	
	def isP1Winner(p1,p2):
		return p1 > p2

	def combat(player1, player2):
		while True:
			winP1 = isP1Winner(player1[0], player2[0])
			winner = player1 if winP1 else player2
			loser = player2 if winP1 else player1
			winner.rotate(-1)
			winner.append(loser.popleft())

			if not player1 :
				return player2
			if not player2 :
				return player1
			
	winner = combat(player1,player2)
	size = len(winner)
	winner.reverse()
	print('winner score is', sum([ (f+1) * winner[f] for f in range(size)]))
	
	
test = '''Player 1:
9
2
6
3
1

Player 2:
5
8
4
7
10'''

main(lines)

File: 2022/day2/1.js
const fs = require('fs');
const input = fs.readFileSync('./input.txt', 'utf8');

const test = `A Y
B X
C Z`;

const getScore = ([a, b]) => {
  let base = 0, score = 0;
  
  switch (b) {
    case 'X':
      base = 1;
      break;
    case 'Y':
      base = 2;
      break;
    case 'Z':
      base = 3;
      break;
  }
  switch (a) {
    case 'A':
      if (b=='Y') score = 6;
      if (b=='X') score = 3;
      break;
    case 'B':
      if (b=='Z') score = 6;
      if (b=='Y') score = 3;
      break;
    case 'C':
      if (b=='X') score = 6;
      if (b=='Z') score = 3;
      break;
  }
  return base + score
}

const main1 = (str) => {
  const series = str.split('\n');
  return series.reduce((sum, s) => {
    return sum + getScore(s.split(' '));
  }, 0);
}

console.log(main1(test));
console.log(main1(input));
